code/game/machinery/telecomms/machine_interactions.dm

```diff
@@ -14,6 +14,23 @@
 	var/temp = "" // output message
 	var/construct_op = 0
 
+/obj/machinery/telecomms/examine(mob/user)
+	..(user)
+	switch(integrity)
+		if(100)
+			user << SPAN_NOTICE("It's in perfect condition.")
+		if(85 to 99)
+			user << "It has some minor faults."
+		if(65 to 85)
+			user << "It's slightly damaged."
+		if(45 to 65)
+			user << SPAN_WARNING("It's badly damaged.")
+		if(25 to 45)
+			user << SPAN_WARNING("It's heavily damaged.")
+		else
+			user << SPAN_WARNING("It's falling apart.")
+	if(stat & BROKEN)
+		user << SPAN_DANGER("It's broken.")
```

Comment:
Isn't it better to check `stat` first and if it's not `BROKEN` check `integrity`?
That seems right now you can get message like _"This is telecomms machinry. It's in perfect condition It's broken."_

Reply:
Tcomms machinery has two separate damage conditions: integrity, which is caused by overheating, and the BROKEN stat, which only happens when you remove cables from it while disassembling it.

I have changed the broken message to better reflect its status.